decidim-elections/app/uploaders/decidim/votings/census/voting_census_uploader.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Votings
+    module Census
+      # This class deals with saving voting cenusus access codes export Zip Files to App
+      class VotingCensusUploader < Decidim::ApplicationUploader
+        # Override the directory where uploaded files will be stored.
+        def store_dir
+          default_path = "uploads/voting-census/"
+
+          return File.join(Decidim.base_uploads_path, default_path) if Decidim.base_uploads_path.present?
```

Comment:
Is this path accessible by the public or you need to be logged in as an admin? If it's using something external as S3 we probably need some kind of access token.

Reply:
In the email we are sending the admin an URL pointing the the voting's admin area (hence they need to be logged in).
From there, they'll be [redirected to the file URL](https://github.com/decidim/decidim/pull/7756/files#diff-9b0deb4e99860caba6da634bded04e8a258bf078082b3e67db58c0ffa9815d66R94).

But yes, the file URL itself is public. The filename is a randomly generated alphanumeric string and the zip file is password-protected.
Just as a reference, this is the same flow we are using for the data portability export (https://github.com/decidim/decidim/pull/5342)